File: Program/ActEditor/src/pl/wroc/pwr/student/acteditor/model/tags/SimpleElement.java
package pl.wroc.pwr.student.acteditor.model.tags;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Przechowuje i udostepnia dane o elemencie dokumentu XML.
 * 
 * @author Mateusz
 * 
 */
public class SimpleElement implements Element {
	private List attributes = Collections.synchronizedList(new ArrayList());
	private String minOccurs = "1";
	private String maxOccurs = "1";
	private String name = "";
	private String type = "";
	private String description = "";
	private String value = "";

	/**
	 * Tworzy obiekt typu SimpleElement z ustawiona wartoscia name.
	 * 
	 * @param name
	 *          Nazwa elementu.
	 */
	public SimpleElement(String name) {
		this.name = name;
	}

	public String toString() {
		return name + " " + description + " " + minOccurs + "/" + maxOccurs;
	}

	@Override
	public void add(Element element) {
		// unsupported
	}

	@Override
	public void remove(Element element) {
		// unsupported
	}

	@Override
	public Element get(int index) {
		// unsupported
		return null;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	@Override
	public List getElements() {
		return null;
	}

	@Override
	public String getDescription() {
		return description;
	}

	@Override
	public void setDescription(String description) {
		this.description = description;
	}

	@Override
	public String getType() {
		return type;
	}

	@Override
	public void setType(String type) {
		this.type = type;
	}

	@Override
	public List getAttributes() {
		return attributes;
	}

	@Override
	public void addAttribute(Attribute attribute) {
		attributes.add(attribute);
	}

	@Override
	public String getMinOccurs() {
		return minOccurs;
	}

	@Override
	public void setMinOccurs(String minOccurs) {
		this.minOccurs = minOccurs;
	}

	@Override
	public String getMaxOccurs() {
		return maxOccurs;
	}

	@Override
	public void setMaxOccurs(String maxOccurs) {
		this.maxOccurs = maxOccurs;
	}

	@Override
	public String getValue() {
		return value;
	}

	@Override
	public void setValue(String value) {
		this.value = value;
	}

}

File: Program/ActEditor/bin/pl/wroc/pwr/student/acteditor/model/tags/SimpleElement.js
Clazz.declarePackage ("pl.wroc.pwr.student.acteditor.model.tags");
Clazz.load (["pl.wroc.pwr.student.acteditor.model.tags.Element", "java.util.ArrayList", "$.Collections"], "pl.wroc.pwr.student.acteditor.model.tags.SimpleElement", null, function () {
c$ = Clazz.decorateAsClass (function () {
this.attributes = null;
this.minOccurs = "1";
this.maxOccurs = "1";
this.name = "";
this.type = "";
this.description = "";
this.value = "";
Clazz.instantialize (this, arguments);
}, pl.wroc.pwr.student.acteditor.model.tags, "SimpleElement", null, pl.wroc.pwr.student.acteditor.model.tags.Element);
Clazz.prepareFields (c$, function () {
this.attributes = java.util.Collections.synchronizedList ( new java.util.ArrayList ());
});
Clazz.makeConstructor (c$, 
function (name) {
this.name = name;
}, "~S");
Clazz.overrideMethod (c$, "toString", 
function () {
return this.name + " " + this.description + " " + this.minOccurs + "/" + this.maxOccurs;
});
Clazz.overrideMethod (c$, "add", 
function (element) {
}, "pl.wroc.pwr.student.acteditor.model.tags.Element");
Clazz.overrideMethod (c$, "remove", 
function (element) {
}, "pl.wroc.pwr.student.acteditor.model.tags.Element");
Clazz.overrideMethod (c$, "get", 
function (index) {
return null;
}, "~N");
Clazz.overrideMethod (c$, "getName", 
function () {
return this.name;
});
Clazz.overrideMethod (c$, "setName", 
function (name) {
this.name = name;
}, "~S");
Clazz.overrideMethod (c$, "getElements", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getDescription", 
function () {
return this.description;
});
Clazz.overrideMethod (c$, "setDescription", 
function (description) {
this.description = description;
}, "~S");
Clazz.overrideMethod (c$, "getType", 
function () {
return this.type;
});
Clazz.overrideMethod (c$, "setType", 
function (type) {
this.type = type;
}, "~S");
Clazz.overrideMethod (c$, "getAttributes", 
function () {
return this.attributes;
});
Clazz.overrideMethod (c$, "addAttribute", 
function (attribute) {
this.attributes.add (attribute);
}, "pl.wroc.pwr.student.acteditor.model.tags.Attribute");
Clazz.overrideMethod (c$, "getMinOccurs", 
function () {
return this.minOccurs;
});
Clazz.overrideMethod (c$, "setMinOccurs", 
function (minOccurs) {
this.minOccurs = minOccurs;
}, "~S");
Clazz.overrideMethod (c$, "getMaxOccurs", 
function () {
return this.maxOccurs;
});
Clazz.overrideMethod (c$, "setMaxOccurs", 
function (maxOccurs) {
this.maxOccurs = maxOccurs;
}, "~S");
Clazz.overrideMethod (c$, "getValue", 
function () {
return this.value;
});
Clazz.overrideMethod (c$, "setValue", 
function (value) {
this.value = value;
}, "~S");
});
